anchor/src/logic.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{token::BLamports, ANCHOR_MINT_AUTHORITY};
+use lido::error::LidoError;
+use solana_program::{
+    account_info::AccountInfo, borsh::try_from_slice_unchecked, entrypoint::ProgramResult, msg,
+    program::invoke_signed, program_error::ProgramError, pubkey::Pubkey,
+};
+
+use crate::state::Anchor;
+
+pub fn deserialize_anchor(
+    program_id: &Pubkey,
+    anchor: &AccountInfo,
+) -> Result<Anchor, ProgramError> {
+    if anchor.owner != program_id {
+        msg!(
+            "Lido state is owned by {}, but should be owned by the Lido program ({}).",
+            anchor.owner,
+            program_id
+        );
+        return Err(LidoError::InvalidOwner.into());
+    }
+    let lido = try_from_slice_unchecked::<Anchor>(&anchor.data.borrow())?;
+    Ok(lido)
```

Comment:
If we go with the idea of letting the Anchor instance live at a program-derived address derived from the Lido instance address, then this would be the place to verify.